### PR TITLE
KIALI-1560 Allow filtering lists by health status

### PR DESCRIPTION
**Pull request by jotak**
Shared function "filterByHealth" can be called in promises chain when filtering by health
Some DRY-mutualisation on filters (common patterns in CommonFilters.ts)

Note, I based this PR on #771 , not that I had no choice but it was easier for me. So ideally, #771 should be reviewed & merged first, then  I can rebase this one.

### CODE REVIEW

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#275](https://paste.fedoraproject.org/paste/MfmTRbxAcLzgjfFqM0bvMg)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1782]](https://paste.fedoraproject.org/paste/aOpWsEQAqf8bbKRO0SaenQ)
  * :heavy_check_mark: Istio-Upstream [#[627]](https://paste.fedoraproject.org/paste/ezWCrTmMH3TfBJLyCqBsJw)
  * :heavy_check_mark: Kiali-Istio-Checker [#[2116, 2117]](https://paste.fedoraproject.org/paste/1XOfVV54rqzq3HjT2uREyg)
  * :heavy_check_mark: install-bookinfo [#[1159]](https://paste.fedoraproject.org/paste/Ks3~igp68eIm8c-xsZWz7w)
  * :heavy_check_mark: Kiali-on-Openshift [#[1156]](https://paste.fedoraproject.org/paste/k-O2rimdUzD9aBct6SD7gQ)
  * :heavy_check_mark: run-kiali-e2e-tests [#[560]](https://paste.fedoraproject.org/paste/aj8QSpaNoVRiTBCh22X0Pw)


KIALI_NAME=kialiuipr773-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#281](https://paste.fedoraproject.org/paste/UFgOIFNwuKdoPBzac4Wbag)
  * :heavy_check_mark: Istio-and-Kiali-on-Openshift [#[1827]](https://paste.fedoraproject.org/paste/0eamgZtDHwLGe4h23vcnNA)
  * :heavy_check_mark: Istio-Upstream [#[670]](https://paste.fedoraproject.org/paste/3sCbkVkl-o-HMae8ouPT6w)
  * :heavy_check_mark: Kiali-Istio-Checker [#[2209, 2210]](https://paste.fedoraproject.org/paste/0WASTGHieeT9a6ngEP9vOQ)
  * :heavy_check_mark: install-bookinfo [#[1207]](https://paste.fedoraproject.org/paste/UQkTzJnxRNahR23Naf7o6A)
  * :heavy_check_mark: Kiali-on-Openshift [#[1204]](https://paste.fedoraproject.org/paste/ti1NPxV9jdgsy6IrE0JS9w)
  * :heavy_check_mark: run-kiali-e2e-tests [#[606]](https://paste.fedoraproject.org/paste/ewBL~untUDEj~dHgMBUR1A)


KIALI_NAME=kialiuipr773-istio-system

**Comment by rhqci** [comment]
:heavy_check_mark: Jenkins CI: kiali-ui-pr-e2e-test [#293](https://paste.fedoraproject.org/paste/8-cp0empxlbt0nLZJRsl6w)
  * :red_circle: Istio-and-Kiali-on-Openshift [#[1853]](https://paste.fedoraproject.org/paste/jzLubZfZxHID8hFeBxW2ig)
  * :heavy_check_mark: Istio-Upstream [#[695]](https://paste.fedoraproject.org/paste/yI-LvicZe2NgjlwWtipWDQ)
  * :heavy_check_mark: Kiali-Istio-Checker [#[2251]](https://paste.fedoraproject.org/paste/NC~2Q4bvioYGYUDyKGg5UA)
  * :red_circle: install-bookinfo [#[1237]](https://paste.fedoraproject.org/paste/IknrtAhUvz4OirVeIrPXfg)


KIALI_NAME=kialiuipr773-istio-system

**Comment by lucasponce** [comment]
I think this one needs a rebase with master, too.

**Comment by lucasponce** [comment]
Adding @xeviknal to review this one today.

**Comment by rhqci** [comment]
Jenkins CI: kiali-ui-pr-e2e-test [#323](https://paste.fedoraproject.org/paste/2aRTTqlQ1iomFhPkk9Td5g)
 :red_circle: PRT Failed, Please Contact QE

File: src/pages/IstioConfigList/IstioConfigListComponent.tsx [hunk 1]
```diff
@@ -93,24 +93,13 @@ class IstioConfigListComponent extends ListComponent.Component<
...
-      .filter(activeFilter => activeFilter.category === 'Config')
-      .map(activeFilter => activeFilter.value);
-
-    /** Remove duplicates  */
-    namespacesSelected = removeDuplicatesArray(namespacesSelected);
-    istioTypeFilters = removeDuplicatesArray(istioTypeFilters);
-    istioNameFilters = removeDuplicatesArray(istioNameFilters);
-    configValidationFilters = removeDuplicatesArray(configValidationFilters);
+    const namespacesSelected = getFilterSelectedValues(IstioConfigListFilters.namespaceFilter, activeFilters);
+    const istioTypeFilters = getFilterSelectedValues(IstioConfigListFilters.istioTypeFilter, activeFilters);
```
[the remark this answers is not in the document]
Comment (jotak) [reply src/pages/IstioConfigList/IstioConfigListComponent.tsx]: thanks for bringing that up, I'm looking at it

[the remark this answers is not in the document]
Comment (jotak) [reply src/pages/IstioConfigList/IstioConfigListComponent.tsx]: @xeviknal I went too fast here and missed the `dicIstioType` that was there before.
Added it back, I've just tested filtering by istio type, seems to work correctly now.

**Comment by rhqci** [comment]
Jenkins CI: kiali-ui-pr-e2e-test [#325](https://paste.fedoraproject.org/paste/pkhLVRFecvy9522cz7eF5A)
 :red_circle: PRT Failed, Please Contact QE

**Comment by xeviknal** [comment]
Tested in my local environment. I do performed the following test for each list page (app/workload/services/config):

- filter for open text value (e.g. name = reviews)
- filter for Presence value (e.g. sidecar = Present)
- filter for Health
- clear filters resets pagination (sets current page to 1)
- adding same filters twice (not allowed)
- adding opposite value for non-appendable filter (swaps value)

- sort by each order field.
- reverse sorting direction.
- app keeps valid filters when switch list page.


Overview page:
- filter by name and health
- sorting by name and status (also reverse order)

File: src/pages/IstioConfigList/IstioConfigListComponent.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (xeviknal) [reply src/pages/IstioConfigList/IstioConfigListComponent.tsx]: @jotak now, it works! Cool :)

**Pull request merged**